client/verta/tests/unit_tests/pipeline/test_pipeline_step.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for the PipelineStep class
+"""
+
+import random
+from unittest.mock import patch
+
+from hypothesis import given, HealthCheck, settings, strategies as st
+
+import verta
+from tests.unit_tests.strategies import pipeline_definition
+from verta.pipeline import PipelineStep
+
+
+@given(
+    pipeline_definition=pipeline_definition(),
+    registered_model_id=st.integers(min_value=1, max_value=2**63),
+    # max value limit avoids protobuf "Value out of range" error
+    model_version_name=st.text(min_size=1),
+    model_name=st.text(min_size=1),
+)
+@settings(
+    suppress_health_check=[HealthCheck.function_scoped_fixture],
+    deadline=None,
+)
+def test_steps_from_pipeline_definition(
+    pipeline_definition,
+    registered_model_id,
+    model_version_name,
+    model_name,
+    mock_conn,
+    mock_config,
+    mocked_responses,
+) -> None:
+    """Test that a list of PipelineStep objects can be constructed and
+    returned from a pipeline definition.
+
+    The registered model, and registered model version is fetched for
+    each step, so a call is mocked for each.
+    """
+    graph = pipeline_definition["graph"]
+    for step in pipeline_definition["steps"]:
+        mocked_responses.get(
+            f"https://test_socket/api/v1/registry/model_versions/{step['model_version_id']}",
+            json={
+                "model_version": {
+                    "id": step["model_version_id"],
+                    "registered_model_id": registered_model_id,
+                    "version": model_version_name,
+                }
+            },
+            status=200,
+        )
+        mocked_responses.get(
+            f"https://test_socket/api/v1/registry/registered_models/{registered_model_id}",
+            json={
+                "registered_model": {
+                    "id": registered_model_id,
+                    "name": model_name,
+                }
+            },
+            status=200,
+        )
+    generated_steps = PipelineStep._steps_from_pipeline_definition(
+        pipeline_definition=pipeline_definition,
+        conn=mock_conn,
+        conf=mock_config,
+    )
+    # we have the same number of steps as in the pipeline definition
+    assert len(generated_steps) == len(pipeline_definition["steps"])
+    # sort both group of steps for side-by-side comparison
+    generated_steps_sorted = sorted(list(generated_steps), key=lambda x: x.name)
```

Comment:
fyi, don't need to cast to `list`

```suggestion
    generated_steps_sorted = sorted(generated_steps, key=lambda x: x.name)
```